src/authorization/WebAclAuthorizer.ts

```diff
@@ -184,20 +184,28 @@ export class WebAclAuthorizer extends Authorizer {
    *
    * @returns A store containing the relevant acl triples.
    */
-  private async getAclRecursive(id: ResourceIdentifier, recurse?: boolean): Promise<Store> {
+  private async getAclRecursive(id: ResourceIdentifier, originalId?: ResourceIdentifier): Promise<Store> {
+    const target = originalId ?? id;
     this.logger.debug(`Trying to read the direct ACL document of ${id.path}`);
     try {
       const acl = this.aclStrategy.getAuxiliaryIdentifier(id);
       this.logger.debug(`Trying to read the ACL document ${acl.path}`);
       const data = await this.resourceStore.getRepresentation(acl, { type: { [INTERNAL_QUADS]: 1 }});
       this.logger.info(`Reading ACL statements from ${acl.path}`);
-
-      return this.filterData(data, recurse ? ACL.default : ACL.accessTo, id.path);
+      const [ direct, recursive ] = await Promise.all([
+        this.filterData(data, ACL.accessTo, target.path),
+        this.filterData(data, ACL.default, id.path),
+      ]);
+
+      return new Store([
+        ...direct.getQuads(null, null, null, null),
+        ...recursive.getQuads(null, null, null, null),
+      ]);
```

Comment:
This is the only place where `filterData` gets called, so I would just change that function instead. This way the store logic is kept out of the `getAclRecursive` function. `filterData` could take an array of predicates as input instead for example. Probably would also make it more efficient as calling `filterData` twice does some duplicate work.

Reply:
Done. Any reason filterdata isn't in some util library, btw?

Reply:
No real reason tbh. Just that functionality this was written for this class and not really needed elsewhere for now.